Implementation of custom behavior for heroes

Badulundabad opened the pull request:
- removed some campaign patches;
- removed UrbanCharacterCampaignBehavior and HeroSpawnCampaginBehavior;
- added TORUrbanCharacterCampaignBehavior that, if shortly, creates notables and companions, and controls behavior of notables;
- added TORHeroSpawnCampaginBehavior that controls behavior of companions;
- added some extensions for Hero and Settlement.

Badulundabad committed:
Added some extension methods

Badulundabad committed:
Added new custom behaviors

- UrbanCharactersCampaignBehavior and HeroSpawnCampaignBehavior are replaced by custom ones

Badulundabad committed:
removed excess patches

Badulundabad committed:
Adjusted behaviors for notables and heroes

hunharibo commented on CSharpSourceCode/Utilities/Extensions/SettlementExtensions.cs:
```diff
@@ -33,5 +33,19 @@ public static List<string> GetSceneNames(this Settlement settlement)
+                   (settlement.MapFaction.Name.Contains("Moot") ||
+                    settlement.MapFaction.Name.Contains("Averland") ||
+                    settlement.MapFaction.Name.Contains("Stirland"));
+        }
```
I suggest using the Culture instead of Faction Name. You can access Culture directly from settlement through a property.

Z3rca replied:
agree

Badulundabad replied:
Settlement doesn't change its culture when it's conquered, so just imagine empire settlement with vampire culture. Currently my solution is pretty acceptable i think, but then we'll rework it for sure.

hunharibo reviewed (commented):
good work, one remark

Badulundabad committed:
Merge branch 'development' into feature/NotablesAndWnaderersBehaviorImplementation

Z3rca reviewed (commented):
from my side, when you update it to the new version we can use that. Also try to make your pull request maybe a bit more informative. The description text could be a bit more informative.

Badulundabad committed:
Replaced behaviors by new ones

- replaced CustomHeroSpawnCampaignBehavior and CustomUrbanCharactersCampaignBehavior by new ones;
- removed related campaign harmony patches;

Badulundabad committed:
adjusted urban character behavior

Badulundabad committed:
Merge branch 'development' into feature/NotablesAndWnaderersBehaviorImplementation
